Newest update from video_streaming group

Mikkelmmc opened the pull request:
## Description

This is a pull request from the video streaming group, where we worked on the explore screen and the navigation bar in the bottom

## Changes

Created:
- Navigation bar (bottom)
- Explore screen
- Explore course card

## Related Issues

No. 

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)

<img width="220" alt="Skærmbillede 2023-10-06 kl  11 52 04" src="https://github.com/Educado-App/educado-mobile/assets/41455260/7c515579-01e4-47e2-92ca-ef7cc3410b30">

<img width="220" alt="Skærmbillede 2023-10-06 kl  11 52 23" src="https://github.com/Educado-App/educado-mobile/assets/41455260/1f8caa01-535d-4c8a-8c59-83b3d3c8a544">

<img width="220" alt="Skærmbillede 2023-10-06 kl  11 51 31" src="https://github.com/Educado-App/educado-mobile/assets/41455260/fa0f727b-1f82-482a-af0a-5887d7264e7c">




## Notes for Reviewers

It's mostly changes in the design, it is working on both IOS and Android

SlovenlyTF commented:
Husk at bruge den pull request template for projectet :D
https://github.com/Educado-App/resources/blob/main/PULL_REQUEST_TEMPLATE.md

SlovenlyTF commented:
Er den her pull request en færdig PBI, eller bare kode der virker til showcase mandag?

Jatewo commented:
Please update the pull request to use the pull request template from the resource repo.

Jatewo commented:
Also remember that you can request reviews from others 👍 

In case you are not ready for reviewing, please convert the PR to a draft 😄

Mikkelmmc commented:
> * Navbar
> * Explorar screen

Mikkelmmc commented:
* [Explore Page + Navbar]
    * We have improved the Navbar as well as the explore page in the mobile app to look like their Figma prototypes. 
🔧 Improvements
* [Improvement Navbar]
    * We fixed the Navbar to look as described in its Figma prototype, and have navigation work properly for existing modules.
* [Improvement Explore Page]
    * We fixed the explore page to look as described in its Figma prototype, and it now has courses to explore within it, that can be folded out to explore further details. 
🐛 Bug Fixes
* [Issue or Bug Name]
    * No particular bugs, but rather just an improvement to what was before. 
📖 Documentation
* “”

JackSMunn commented:
Not using template for PR's. Please do so. Converting to draft to not clutter PRs that need immediate attention.  Reopen when ready.

JackSMunn commented on eml/App.js:
```diff
@@ -98,116 +99,31 @@ function LoginStack() {
-      /> */}
-    </Tab.Navigator>
-  );
-}
```
What is the reason for deleting the whole stack?

Mikkelmmc replied:
Because we made a 'NavBar' component instead.

JackSMunn commented on eml/api/api.js:
```diff
@@ -42,15 +40,21 @@ export const getCourseWithAuth = async (courseId) => {
   // TODO: add bearer token to request header and omit /public
-  const res = await axios.get(url + '/api/public/courses');
+  const res = await axios.get(url + '/api/courses');
+  return res.data;
```
Microsoft convention requires endpoint to be singular. (not plural) 
So /api/course

Jatewo replied:
+1

JackSMunn commented on eml/api/api.js:
```diff
@@ -42,15 +40,21 @@ export const getCourseWithAuth = async (courseId) => {
+};
+
+export const getHome = async () => {
+  const res = await axios.get(url + '/api/course/getHome');
```
Microsoft convention for endpoint says that http request types are not used in endpoints. 
So /api/course/home

Jatewo replied:
+1

JackSMunn commented on eml/components/courses/courseCard/CourseCard.jsx:
```diff
@@ -0,0 +1,29 @@
+      });
+    
+      if (!fontsLoaded) {
+        return AppLoading;
```
Do i understand it correctly, that you just return program into an idling animation if the font is not loading?

Jatewo replied:
This is not needed, it is solved with the new Text component.

JackSMunn commented on eml/components/courses/courseHeader/CourseProgress.jsx:
```diff
@@ -9,26 +9,42 @@ export default function CourseProgress({ fracTop, fracBot }) {
+    color: 'black',
+    position: 'absolute',
+    alignSelf: 'center',
+    bottom: -2,
```
This whole stylesheet block needs to be reworked to use NativeWind and not stylesheets, as agreed upon by integration team.

JackSMunn commented on eml/components/courses/courseHeader/CourseTitleIcon.jsx:
```diff
@@ -3,49 +3,62 @@ import { StyleSheet, View, Image } from 'react-native'
   }
 
   let [fontsLoaded] = useFonts({
     VarelaRound_400Regular
```
Correct font? We're using Montserrat

Jatewo replied:
Not needed (see earlier comments)

JackSMunn commented on eml/components/courses/courseHeader/CourseTitleIcon.jsx:
```diff
@@ -3,49 +3,62 @@ import { StyleSheet, View, Image } from 'react-native'
           ellipsizeMode={'tail'}
-          style={{ fontSize: 30, fontFamily: 'VarelaRound_400Regular', color: 'rgb(75,85,99)' }}
+          style={{fontSize: 23, fontFamily: 'VarelaRound_400Regular', color: 'rgb(255,0,0)'}}
         >
```
Wrong font again

JackSMunn commented on eml/components/courses/courseHeader/CourseTitleIcon.jsx:
```diff
@@ -3,49 +3,62 @@ import { StyleSheet, View, Image } from 'react-native'
+        <Text
+          numberOfLines={1}
+          ellipsizeMode={'tail'}
+          style={{fontSize: 16, fontFamily: 'VarelaRound_400Regular', color: 'rgb(0,0,0)'}}
```
Wrong font

JackSMunn commented on eml/components/courses/courseHeader/CourseTitleIcon.jsx:
```diff
@@ -3,49 +3,62 @@ import { StyleSheet, View, Image } from 'react-native'
+    borderBottomLeftRadius: 20,
+    borderTopLeftRadius: 20,
+    overflow: 'hidden'
   },
```
Again, stylesheets, integration team agreed to use NativeWind.

JackSMunn commented on eml/components/courses/section/SectionCard.jsx:
```diff
@@ -0,0 +1,108 @@
+            </Pressable>
+        </View>
+    );
+}
```
Whole file uses inline-styling. Please use NativeWind as the integration team agreed upon.

JackSMunn reviewed (changes requested):
Looks cool, I love when you can see that something is happening.

Please be descriptive in your changes. E.g. "changes to model" Could be "changes made to x,y,z in user model in file: user.js

Please refactor to NativeWind

JackSMunn commented on eml/components/easyDynComponents/EasyDynamicList.jsx:
```diff
@@ -5,23 +5,23 @@ import SectionItem from '../courses/courseBody/SectionItem'
                         />
                     )
-                })}
+                })*/}
```
Expect to use this code? If answer is no, delete dead code

JackSMunn commented on eml/components/explore/AccesCourseButton.jsx:
```diff
@@ -0,0 +1,33 @@
+    </Pressable>
+  );
+};
+
```
Refactor to use NativeWInd

JackSMunn commented on eml/components/explore/CardLabel.jsx:
```diff
@@ -0,0 +1,20 @@
+            <Text style={{ paddingLeft: 3, fontSize: 13, color: color }} >{displayTitle}</Text>
+        </View>
+    );
+}
```
Refactor to use NativeWind

JackSMunn reviewed (changes requested):
I love when something tangible gets produced and it's not just pure backend stuff.
I have some comments regarding documentation, styles and font.

Did you document enough? I understand the code, but I'm not sure other people will in the future.

Refactor all styles to use NativeWind
Change font, i know Jacob & August have created an easy way to do it. 
Be more specific in changes section of pull request template. "changes to what? where, why"

Jatewo commented on eml/api/api.js:
```diff
@@ -61,6 +65,24 @@ export const getPresignedUrl = async (component_id) => {
 };
 
+
+//CREATE FOR TESTING: BUT IT WORKS
```
Is this a leftover test or an actual function?

Jatewo commented on eml/components/courses/CourseBody.jsx:
```diff
@@ -1,6 +1,6 @@
 import React from 'react'
 import { View } from 'react-native'
-import CourseSectionList from './courseBody/CourseSectionList'
+// TODO: Remove this - import CourseSectionList from './courseBody/CourseSectionList'
```
Is this supposed to be removed?

Jatewo commented on eml/components/courses/CourseHeader.jsx:
```diff
@@ -1,30 +1,34 @@
 import React from 'react'
-import { View, StyleSheet } from 'react-native'
+import { View, StyleSheet, Text } from 'react-native'
```
Update to new Text component

Jatewo commented on eml/components/courses/CourseHeader.jsx:
```diff
@@ -1,30 +1,34 @@
 }
 
 
 const styles = StyleSheet.create({
```
Please refactor to NativeWind as defined in our own guidelines

Jatewo commented on eml/components/courses/courseCard/CourseCard.jsx:
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { View, Text, Image, Pressable, Dimensions } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
+import { useFonts, VarelaRound_400Regular } from '@expo-google-fonts/dev';
```
Not needed anymore

Jatewo commented on eml/components/courses/courseCard/CourseCard.jsx:
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { View, Text, Image, Pressable, Dimensions } from 'react-native';
```
Update to new Text component

Jatewo commented on eml/components/courses/courseCard/CourseCard.jsx:
```diff
@@ -0,0 +1,29 @@
+export default function CourseCard({ course, downloadState }) {
+    const navigation = useNavigation();
+    
+    let [fontsLoaded] = useFonts({
```
Not needed anymore

Jatewo commented on eml/components/courses/courseCard/CourseCard.jsx:
Use NativeWind

Jatewo commented on eml/components/courses/courseHeader/CourseProgress.jsx:
```diff
@@ -1,5 +1,5 @@
 import React from 'react'
-import { StyleSheet, View, Text } from 'react-native'
+import { StyleSheet, View, Text} from 'react-native'
```
Update to new Text component

Jatewo commented on eml/components/courses/courseHeader/CourseProgress.jsx:
Use NativeWind

Jatewo commented on eml/components/courses/courseHeader/CourseTitleIcon.jsx:
```diff
@@ -3,49 +3,62 @@ import { StyleSheet, View, Image } from 'react-native'
 import Text from '../../general/Text';
 import PropTypes from 'prop-types'
 import { useFonts, VarelaRound_400Regular } from '@expo-google-fonts/dev'
```
Not needed

Jatewo commented on eml/components/courses/courseHeader/CourseTitleIcon.jsx:
Use NativeWind

Jatewo commented on eml/components/courses/section/SectionCard.jsx:
NativeWind

Jatewo commented on eml/components/courses/section/SectionCard.jsx:
```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react';
+import { View, Text, Pressable, Image, TouchableOpacity } from 'react-native';
```
Update to new Text component (won't comment on further files, but please change this for all 😄)

Jatewo commented on eml/components/easyDynComponents/easyDynCard.jsx:
NativeWind (won't comment anymore about this but please fix for all files 😄)

Jatewo commented on eml/components/navBar/NavBar.jsx:
```diff
@@ -1,15 +1,163 @@
+    VarelaRound_400Regular
+  })
+
+  if (!fontsLoaded) {
```
Not needed

Jatewo commented on eml/components/navBar/NavBar.jsx:
```diff
@@ -1,15 +1,163 @@
 export default function NavBar() {
+
+  let [fontsLoaded] = useFonts({
+    VarelaRound_400Regular
```
Not needed

Jatewo commented on eml/components/navBar/NavBar.jsx:
```diff
@@ -1,15 +1,163 @@
+import TestScreen from '../../screens/test/TestScreen';
+import ProfileComponent from '../../screens/profile/Profile';
+import { Icon } from '@rneui/themed';
+import { useFonts, VarelaRound_400Regular } from '@expo-google-fonts/dev'
```
Not needed

Jatewo commented on eml/components/navBar/NavBar.jsx:
NativeWind

Jatewo commented on eml/package.json:
```diff
@@ -40,10 +40,11 @@
     "bootstrap": "^5.2.1",
     "expo": "^46.0.15",
-    "expo-app-loading": "^2.1.1",
+    "expo-app-loading": "~2.1.0",
```
Are we supposed to use an older version?

Jatewo reviewed (changes requested):
Overall good work. However, you are not using NativeWind in the files as described in our guidelines. Furthermore the new Text component should be used which handles the whole font import issue. Please respond to any comments if you are unsure about anything. Feel free to re-request my review when the requested changes has been addressed.

JackSMunn commented:
Ping me when conflicts are resolved and checks pass, then I will review.

JackSMunn commented:
Closing this pull request. As last commit was 10th october.
